modules/screenshots/module.php

```diff
@@ -0,0 +1,180 @@
+<?php
+
+namespace Elementor\Modules\Screenshots;
+
+use Elementor\Core\Files\CSS\Post_Preview;
+use Elementor\Core\Base\Module as BaseModule;
+use Elementor\User;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly
+}
+
+class Module extends BaseModule {
+
+	public function get_name() {
+		return 'screenshots';
+	}
+
+	public function screenshot_proxy() {
+		if ( ! wp_verify_nonce( $_GET['nonce'], 'screenshot_proxy' ) ) {
+			echo '';
+		}
+
+		if ( ! empty( $_GET['href'] ) ) {
+			$response = wp_remote_get( utf8_decode( $_GET['href'] ) );
+			$body = wp_remote_retrieve_body( $response );
+			if ( $body ) {
+				echo $body;
+			}
+		}
+	}
+
+	public function ajax_save( $data ) {
+		if ( empty( $data['screenshot'] ) ) {
+			return false;
+		}
+
+		$post_id = $data['post_id'];
+
+		$file_content = substr( $data['screenshot'], strlen( 'data:image/png;base64,' ) );
+		$file_name = 'Elementor Post Screenshot ' . $post_id . '.png';
```

Comment:
I think your preview fix was good:
```suggestion
		$file_name = 'Elementor-post-screenshot-' . $post_id . '.png';
```